libtelemetry/src/main/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -28,8 +24,7 @@
 import okhttp3.Response;
 import okhttp3.ResponseBody;
 
-public class MapboxTelemetry implements FullQueueCallback, EventCallback, ServiceTaskCallback,
-  LifecycleObserver {
+public class MapboxTelemetry implements FullQueueCallback, EventCallback, ServiceTaskCallback {
```

Comment:
since you're removing ` LifecycleObserver` this change is breaking public APIs right?

Reply:
I actually will not have to do that. I have come up with a much better and effective solution. Just pulled in master to get the code coverage for planning out tests.